lib/plugins/aws/provider/awsProvider.js

```diff
@@ -222,21 +222,114 @@ class AwsProvider {
             required: ['Fn::Sub'],
             additionalProperties: false,
           },
+          awsLambdaEnvironment: {
+            type: 'object',
+            patternProperties: {
+              '^[A-Za-z_][a-zA-Z0-9_]*$': {
+                anyOf: [
+                  { type: 'string' },
+                  { $ref: '#/definitions/awsCfImport' },
+                  { $ref: '#/definitions/awsCfRef' },
+                ],
+              },
+            },
+            additionalProperties: false,
+          },
+          awsLambdaLayers: {
+            type: 'array',
+            items: { $ref: '#/definitions/awsArn' },
+          },
+          awsLambdaMemorySize: { type: 'integer', multipleOf: 64, minimum: 128, maximum: 3008 },
+          awsLambdaPackage: {
+            type: 'object',
+            properties: {
+              include: { type: 'array', items: { type: 'string' } },
+              exclude: { type: 'array', items: { type: 'string' } },
+              excludeDevDependencies: { type: 'boolean' },
+              artifact: { type: 'string' },
+              individually: { type: 'boolean' },
```

Comment:
I think `excludeDevDependencies` and `individually` should only be supported top-level.

Also I think we don't need a `awsLambdaPackage` definition, as it serves just `functions[]` context (top-level `package` is in `configSchema.js` and not applicable for extensions)